include/deal.II/multigrid/mg_transfer_global_coarsening.templates.h

```diff
@@ -4089,9 +4190,26 @@ MGTwoLevelTransferNonNested<dim, LinearAlgebra::distributed::Vector<Number>>::
       }
   };
 
-  rpe.template process_and_evaluate<Number>(evaluation_point_results,
-                                            buffer,
-                                            evaluation_function);
+  rpe.template process_and_evaluate<value_type>(evaluation_point_results,
+                                                buffer,
+                                                evaluation_function);
+}
+
+
+
+template <int dim, typename Number>
+void
+MGTwoLevelTransferNonNested<dim, LinearAlgebra::distributed::Vector<Number>>::
+  restrict_and_add_internal(
+    LinearAlgebra::distributed::Vector<Number> &      dst,
+    const LinearAlgebra::distributed::Vector<Number> &src) const
+{
+  if (this->fe_coarse->n_components() == 1)
+    restrict_and_add_internal_comp<1>(dst, src);
+  else if (this->fe_coarse->n_components() == dim)
+    restrict_and_add_internal_comp<dim>(dst, src);
+  else
+    AssertThrow(false, ExcNotImplemented());
```

Comment:
Once https://github.com/dealii/dealii/pull/15163 is merged, we can move the conversion of template arguments more inside...